Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -199,18 +191,9 @@ define([
          * @type {Boolean}
          * @default true
          */
-        this.enablePickFeatures = defaultValue(options.enablePickFeatures, true);
+        this.enablePickFeatures = true;
 
-        var credit = options.credit;
-        if (typeof credit === 'string') {
-            credit = new Credit(credit);
-        }
-        this._credit = credit;
-
-        this._urlParts = urlTemplateToParts(this._url, tags);
-        this._pickFeaturesUrlParts = urlTemplateToParts(this._pickFeaturesUrl, pickFeaturesTags);
-
-        this._readyPromise = when.resolve(true);
+        this._readyPromise = this.reinitialize(options);
```

Comment:
`this.reinitialize` should assign `_readyPromise` itself, rather than assigning it here.  Otherwise, when someone calls `reinitialize` later, `_readyPromise` will not be set.